src/main/java/be/coronalert/statistics/vaccinations/VaccinationsPoller.java

```diff
@@ -0,0 +1,45 @@
+package be.coronalert.statistics.vaccinations;
+
+import be.coronalert.statistics.config.StatisticsServiceConfig;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Map;
+
+@Component
+public class VaccinationsPoller {
+  @Autowired
+  private StatisticsServiceConfig statisticsServiceConfig;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  /**
+   * Retrieves the sciensano vaccinations data.
+   *
+   * @return A map containing the amount of vaccinations for dose 1 and 2.
+   */
+  public Map<VaccinationLevel, Integer> pollResults() throws IOException {
+    URL url = new URL(statisticsServiceConfig.getVaccinations().getUrl());
+
+    Vaccination[] entries = objectMapper.readValue(url, Vaccination[].class);
+
+    Integer doseC = getNumberOfVaccinationsForDose(entries, "C");
+    Integer atLeastPartiallyVaccinated = getNumberOfVaccinationsForDose(entries, "A") + doseC;
+    Integer fullyVaccinated = getNumberOfVaccinationsForDose(entries, "B") + doseC;
+
+    return Map.of(VaccinationLevel.PARTIALLY, atLeastPartiallyVaccinated, VaccinationLevel.FULLY, fullyVaccinated);
+  }
+
+  private Integer getNumberOfVaccinationsForDose(Vaccination[] entries, String dose) {
```

Comment:
Think all of this can be done in one shot through the grouping by dose and then summarizing the count in each group : 

Arrays.stream(entries).collect(Collectors.groupingBy(Vaccination::getDose, Collectors.summingInt(Vaccination::getCount)))